src/main/scala/sparkz/core/network/NetworkController.scala

```diff
@@ -233,14 +243,30 @@ class NetworkController(settings: NetworkSettings,
   private def scheduleConnectionToPeer(): Unit = {
     context.system.scheduler.scheduleWithFixedDelay(5.seconds, 5.seconds) {
       () => {
-        if (connections.size < settings.maxConnections) {
+        if (canEstablishNewOutgoingConnection) {
           log.debug(s"Looking for a new random connection")
```

Comment:
Let's set log level here as `log.trace`. It produce so much noise in the logs in the real environment.
Anther point to discuss separately: why do we try to connect every 5 seconds, I think we may do it less frequent and consider the cases when we don't have new candidates for connections -> we may increase the delay

Reply:
As we discussed, we're going to move the connection logic changes to a different ticket.